### PR TITLE
Patch kube-vip for tkg 1.6

### PR DESCRIPTION
**Pull request by aartij17**
### What this PR does / why we need it
This PR enables patching of kube-vip for TKG 1.6. 
The following changes are made: 
1. Add `cp_enable` env variable
2. Replace arg `start` with `manager`.
3. Replace capability `SYS_TIME` with `NET_RAW`
4. Add `HostAliases`

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

#### Special notes for your reviewer
From the tests, screenshot of the modified kube-vip pod
image

### CODE REVIEW

**Comment by aartij17** [comment]
@saji-pivotal - Can you please take a look?

File: pkg/v1/tkg/client/kubevip_update_helper.go [hunk 1]
```diff
@@ -51,9 +51,29 @@ func ModifyKubeVipTimeOutAndSerialize(currentKubeVipPod *corev1.Pod, newLeaseDur
 		}
 		envVars = append(envVars, envVar)
 	}
-
+	envVar := corev1.EnvVar{Name: "cp_enable", Value: "true"}
```
Comment (saji-pivotal) [inline pkg/v1/tkg/client/kubevip_update_helper.go:54]: nit-ish: @aartij17 Thanks for these changes. Is it possible to move these changes into a helper function? Or maybe refactor `ModifyKubeVipTimeOutAndSerialize` to something like `ModifyKubeVipAndSerialize` and have that call meaningful helper functions?

Comment (saji-pivotal) [inline pkg/v1/tkg/client/kubevip_update_helper.go:54]: nit: Can we make `cp_enable` a const string and reference that here?

File: pkg/v1/tkg/client/kubevip_update_helper.go [hunk 2]
```diff
@@ -51,9 +51,29 @@ func ModifyKubeVipTimeOutAndSerialize(currentKubeVipPod *corev1.Pod, newLeaseDur
...
+		if capability == "SYS_TIME" {
+			currentKubeVipPod.Spec.Containers[0].SecurityContext.Capabilities.Add[index] = "NET_RAW"
+		}
+	}
+	currentKubeVipPod.Spec.HostAliases = []corev1.HostAlias{
+		{
+			IP:        "127.0.0.1",
+			Hostnames: []string{"kubernetes"},
+		},
+	}
```
Comment (saji-pivotal) [inline pkg/v1/tkg/client/kubevip_update_helper.go:76]: nit: IMO all the hardcoded strings should be made constants, and ought to be referenced if they're already constants. Thoughts?

File: pkg/v1/tkg/client/upgrade_cluster_test.go [hunk 3]
```diff
@@ -795,6 +795,8 @@ var _ = Describe("When upgrading cluster with fake controller runtime client", f
 				Expect(err).To(BeNil())
 				Expect(len(newKCP.Spec.KubeadmConfigSpec.Files)).To(Equal(1))
 				Expect(newKCP.Spec.KubeadmConfigSpec.Files[0].Content).To(ContainSubstring("value: \"30\""))
+				Expect(newKCP.Spec.KubeadmConfigSpec.Files[0].Content).To(ContainSubstring("name: cp_enable"))
```
Comment (saji-pivotal) [inline pkg/v1/tkg/client/upgrade_cluster_test.go:798]: Thanks for adding the test cases!

**Review by saji-pivotal — COMMENTED** [review]
@aartij17 This looks good overall. I do have a few comments

File: pkg/v1/tkg/client/kubevip_update_helper.go (hunk 1 above)
Comment (aartij17) [reply pkg/v1/tkg/client/kubevip_update_helper.go]: Makes sense! I have refactored this part a bit :) PTAL!

Comment (saji-pivotal) [reply pkg/v1/tkg/client/kubevip_update_helper.go]: Thanks!

**Review by saji-pivotal — COMMENTED** [review]
@aartij17 Thanks for the changes! They look good to me

**Comment by aartij17** [comment]
Closing this and opening a new PR - https://github.com/vmware-tanzu/tanzu-framework/pull/1840

**Pull request closed without merging**